ebook/1_latex2html.py

```diff
@@ -0,0 +1,689 @@
+        r"\\label\{(.+?)\}",
+        r'<span class="headline_label">\1</span>',
+        s,
+        flags=re.DOTALL,
+    )
+    s = re.sub(
+        r"\\headline\{(.+?)\}",
+        r'<span class="headline_headline">\1</span>',
+        s,
+        flags=re.DOTALL,
+    )
+
+    # \begin{enumerate} -> ol
+    s = re.sub(
+        r"\\begin\{enumerate\}\[(.)\.\](.+?)\\end\{enumerate\}",
+        r'<ol type="\1">\2</ol>\n',
+        s,
+        flags=re.DOTALL | re.IGNORECASE,
+    )
+    s = re.sub(
+        r"\\begin\{enumerate\}(.+?)\\end\{enumerate\}",
+        r"<ol>\1</ol>\n",
+        s,
+        flags=re.DOTALL | re.IGNORECASE,
+    )
+
+    s = re.sub(r"\s*\\item(.+?)\n", r"<li>\1</li>\n", s)
+
+    # \parsel
+    myMatches = re.finditer(r"(\\parsel\{([^\}\\]+)\})", s)
+    for myMatch in myMatches:
+        was = myMatch.group(1)
+        womit = convert_parsel(myMatch.group(2))
+        s = s.replace(was, womit)
+
+    # \later
+    s = re.sub(
+        r"\\later\b", r'<div class="later">*</div>', s, flags=re.DOTALL | re.IGNORECASE
+    )
+
+    # footnotes_authorsnotetext
+    myMatches = re.finditer(
+        r"(\\authorsnotetext\{([^\}\\]+?)\})", s, flags=re.DOTALL | re.IGNORECASE
+    )
+    for myMatch in myMatches:
+        was = myMatch.group(1)
+        womit = f" [Author's Note: <i>{myMatch.group(2).strip()}</i>] "
+        # womit = convert_footnotes(myMatch.group(2), authorsnote=True)
+        s = s.replace(was, womit)
+
+    # footnotetext
+    myMatches = re.finditer(
+        r"(\\footnotetext\{([^\}\\]+?)\})", s, flags=re.DOTALL | re.IGNORECASE
+    )
+    for myMatch in myMatches:
+        was = myMatch.group(1)
+        womit = f" [Author's Note: <i>{myMatch.group(2).strip()}</i>] "
+        # womit = convert_footnotes(myMatch.group(2), authorsnote=False)
+        s = s.replace(was, womit)
+
+    # leftovers
+    s = re.sub(r"\{\s*\}", r"", s, flags=re.DOTALL)
+    s = re.sub(r"\[\s*\]", r"", s, flags=re.DOTALL)
+
+    # Latex spaces, etc
+    s = s.strip()
+
+    s = s.replace("\n\n", "</p>\n<p>")
+    # fixing p's
+    s = s.replace("<p>\n", "<p>")
+    s = s.replace("</h1></p>", "</h1>")
+    s = s + "</p>\n"
+    s = s.replace("<p><h2>", "<h2>")
+    s = s.replace("</h2></p>", "</h2>")
+    s = s.replace("<p><div ", "<div ")
+    s = s.replace("</div></p>", "</div>")
+
+    s = re.sub(r"</i>\]\s*</p>", "</i>] ", s, flags=re.DOTALL)
+
+    s = s.replace("<p></p>\n", "")
+    # multiple spaces
+    s = re.sub(r"  +", r" ", s)
+    return s
+
+
+def convert_chapter(s: str) -> str:
+    global counter_chapter
+    counter_chapter += 1
+    # chapter class is used in calibre to detect chapters
+    out = f'<h1 class="chapter">{counter_chapter}. {s}</h1>'
+    return out
+
+
+def convert_parsel(s: str) -> str:
+    s = s.replace("ss", "ß").replace("s", "ss").replace("ß", "sss")
+    out = f'<span class="parsel">{s}</span>'
+    return out
+
+
+# def convert_footnotes(s: str, authorsnote: bool = False) -> str:
```

Comment:
Why is this code left commented out here?

Reply:
It was a try to use a footnote feature. I discovered too late, that it only works in epub V3.
At [calibre ](https://manual.calibre-ebook.com/generated/en/ebook-convert.html#epub-output-options) it says 
> The version of the EPUB file to generate. EPUB 2 is the most widely compatible, only use EPUB 3 if you know you actually need it.
So I decided not to use it, and to produce a more compatible EPUB 2 file and use inline author comments instead.
I added a comment to the code.